specs/GetFavicon.md

```diff
@@ -0,0 +1,146 @@
+{
+    string  value = sender.faviconUri;
+    SaveFaviconUrl(url);
+    System.IO.Stream stream = new System.IO.MemoryStream();
+
+    sender.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Jpeg,
+        stream,
+        (HRESULT code) => {
+            SaveBytes(stream.ToArray());
+        }
+    );
+
+
+};
+```
+# API Notes
+If a Web page does not have a Favicon, then the event is not fired.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// Interface for the Favicon changed event handler
+[uuid(A0CDE626-8E5E-4EE2-9766-58F3696978FE), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(93ACC5AD-DC22-419E-9A3F-75D96A1538E4), object, pointer_default(unique)]
+interface ICoreWebView2GetFaviconCompletedHandler : IUnknown {
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke([in] HRESULT error_code);
+}
+
+[uuid(DC838C64-F64B-4DC7-98EC-0992108E2157), object, pointer_default(unique)]
+interface ICoreWebView2_5 : ICoreWebView2_4 {
+  /// Add an event handler for the `FaviconChanged` event.
+  /// `FaviconChanged` runs when the WebView favicon changes
+  HRESULT add_FaviconChanged(
+        [in] ICoreWebView2FaviconChangedEventHandler* eventHandler,
+        [out] EventRegistrationToken* token);
+
+  /// Removing the event handler for `FaviconChanged` event
+  HRESULT remove_FaviconChanged(
+      [in] EventRegistrationToken token);
+
+/// Async function for getting the actual image data of the favicon
+ HRESULT GetFaviconAsync(
+        [in] COREWEBVIEW2_FAVICON_IMAGE_FORMAT format,
+        [in] IStream * imageStream,
+        [in] ICoreWebView2GetFaviconCompletedHandler* eventHandler
+ );
+
+  /// used to access the current value of the favicon
+  /// If a page has no favicon then this returns a nullptr
+  [propget] HRESULT FaviconUri([out, retval] LPWSTR* value);
+
+}
+[v1_enum]
+typedef enum COREWEBVIEW2_FAVICON_IMAGE_FORMAT {
+    /// Indicates that the function should return the favicon in PNG format
+    COREWEBVIEW2_CAPTURE_PREVIEW_IMAGE_FORMAT_PNG,
+
+    /// Indicates that the function should return the favicon in JPG format
+    COREWEBVIEW2_CAPTURE_PREVIEW_IMAGE_FORMAT_JPEG,
+}
+```
+
+## .Net/ WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core {
+
+    delegate void CoreWebView2GetFaviconCompletedHandler()
+
+     enum CoreWebView2FaviconImageFormat
+    {
+        Png = 0,
+        Jpeg = 1,
+    };
+
+/// Interface for the Favicon uri changed event handler
+    runtimeclass CoreWebView2 {
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, Object> FaviconChanged;
+        string FaviconUri {get;};
+        void GetFaviconAsync(
```

Comment:
You should be able to copy the MIDL3 content out of Microsoft.Web.WebView2.Core.idl generated by build-apiwriter. The async API should look something like this:

`Windows.Foundation.IAsyncAction CapturePreviewAsync(CoreWebView2CapturePreviewImageFormat imageFormat, Windows.Storage.Streams.IRandomAccessStream imageStream);`

In WinRT there won't be a delegate (above) and you don't pass in a handler parameter